README.md
# body-miscale-card

[![GH-release](https://img.shields.io/github/v/release/dckiller51/lovelace-body-miscale-card.svg?style=flat-square)](https://github.com/ale87jan/lovelace-body-miscale-card/releases)
[![GH-downloads](https://img.shields.io/github/downloads/dckiller51/lovelace-body-miscale-card/total?style=flat-square)](https://github.com/ale87jan/lovelace-body-miscale-card/releases)
[![GH-last-commit](https://img.shields.io/github/last-commit/dckiller51/lovelace-body-miscale-card.svg?style=flat-square)](https://github.com/ale87jan/lovelace-body-miscale-card/commits/main)
[![GH-code-size](https://img.shields.io/github/languages/code-size/dckiller51/lovelace-body-miscale-card.svg?color=red&style=flat-square)](https://github.com/ale87jan/lovelace-body-miscale-card)
[![hacs_badge](https://img.shields.io/badge/HACS-Custom-orange.svg?style=flat-square)](https://github.com/hacs)

Card for data of Xiaomi scales in the Lovelace user interface of Home Assistant

The card is linked to the Bodymiscale custom components for Xiaomi scales. <https://github.com/ale87jan/lovelace-body-miscale-card> based on <https://github.com/dckiller51/bodymiscale>

## Installation

Manually add [body-miscale-card.js](https://raw.githubusercontent.com/dckiller51/lovelace-body-miscale-card/master/dist/body-miscale-card.js)
to your `<config>/www/` folder and add the following to the `configuration.yaml` file:

```yaml
lovelace:
  resources:
    - url: /local/body-miscale-card.js?v=1.0.0
      type: module
```

_OR_ install using [HACS](https://hacs.xyz/) and add this (if in YAML mode):

```yaml
lovelace:
  resources:
    - url: /hacsfiles/lovelace-body-miscale-card/body-miscale-card.js
      type: module
```

The above configuration can be managed directly in the Configuration -> Lovelace Dashboards -> Resources panel when not using YAML mode,
or added by clicking the "Add to lovelace" button on the HACS dashboard after installing the plugin.

If you want to use the scales background image, download and add
[src/images/miscale2.jpg](https://raw.githubusercontent.com/dckiller51/lovelace-body-miscale-card/master/src/images/miscale2.jpg)
to `<config>/www/images/` or configure your own preferred path.

For body score icons, download and add
[src/images/bodyscoreIcon/*.png](https://raw.githubusercontent.com/dckiller51/lovelace-body-miscale-card/master/src/images/bodyscoreIcon)
to `<config>/www/images/bodyscoreIcon/`.

## Configuration

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| type | string | **Required** | `custom:body-miscale-card`
| entity | string | **Required** | `bodymiscale.name`
| name | string/bool | `friendly_name` | Override friendly name (set to `false` to hide)
| image | string/bool | `false` | Set path/filename of background image (i.e. `/local/img/miscale2.jpg`)
| attributes | [Entity Data](#entity data) | *(see below)* | Set to `false` to hide all attributes
| buttons | [Button Data](#button data) | *(see below)* | Set to `false` to hide button row

### Entity Data

Default bodymiscale attributes under each list:

- `attributes` (**right list**) include `weight`, `impedance` (Optional), `height`, `age` and `gender`.
- `body` (**below list**) include `water` (miscale 181B), `visceral_fat`, `body_fat` (miscale 181B), `bmi`, `muscle_mass` (miscale 181B),
                                 `protein` (miscale 181B), `basal_metabolism`, `bone_mass` (miscale 181B), `metabolic_age` (miscale 181B),
                                 `ideal`, `body_type`.

See [examples](#examples) on how to customize, hide or add custom attributes.

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| key | string | **Required** | Attribute key on bodymiscale entity
| icon | string | | Optional icon
| label | string | | Optional label text
| unit | string | | Optional unit

### Button Data

Default buttons include `user1`, `user2`, `user3`, `user4` and `user5`.
See [examples](#examples) on how to customize, hide or add custom buttons/actions.

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| icon | string | **Required** | Show or hide stop button
| service | string | **Required** | Service to call (`input_boolean.toggle`)
| show | bool | `true` | Show or hide button
| label | string | | Optional label on hover
| service_data | object | | entity_id: input_boolean.bodyscale_user1_info_toggle

### Other models

Define your model. `false` (181D) or `true` (181B) (with to impedance)

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| model | bool | `false` | Supported models: with or without impedance

## Screenshots

![body-miscale-card](https://raw.githubusercontent.com/dckiller51/lovelace-body-miscale-card/master/examples/add-card.png)

![body-miscale-card-image](https://raw.githubusercontent.com/dckiller51/lovelace-body-miscale-card/master/examples/card-configuration.png)

## Examples

Basic configuration:

```yaml
- type: custom:body-miscale-card
  entity: bodymiscale.name
```

```yaml
- type: custom:body-miscale-card
  entity: bodymiscale.name
  image: /local/custom/folder/background.jpg
  name: My Bodymiscale
  model: false
```

Hide specific attributes and/or buttons:

```yaml
- type: custom:body-miscale-card
  entity: bodymiscale.name
  attributes:
    age: false
    gender: false
  buttons:
    user1: false
```

Customize specific buttons:

```yaml
- type: custom:body-miscale-card
  entity: bodymiscale.name
  buttons:
    user1:
      icon: 'mdi:alpha-a-circle'
      label: Aurélien
      service_data:
        entity_id: input_boolean.bodyscale_aurelien_info_toggle
```

Add custom attributes:

```yaml
- type: custom:body-miscale-card
  entity: bodymiscale.name
  attributes:
    body_type:
      key: body_type
      label: 'Body type: '
    water:
      key: water
      label: 'Eau: '
      unit: '%'
```

Add custom buttons and service calls:

```yaml
- type: custom:body-miscale-card
  entity: bodymiscale.name
  buttons:
    new_button:
      icon: mdi:light-switch
      label: Custom button!
      service: light.turn_off
      service_data:
        entity_id: light.living_room
```

Add custom bar options (To know the start, destination, color and target values, open your Mi Fit app on your smartphone.)

## Options

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| color | string | var(--score-card-color, var(--ha-card-background)) | Color of the bar.
| height | string | 30px | Defines the height of the bar.
| max | number | 100 | Defines maximum value of the bar.
| min | number | 0 | Defines minimum value of the bar.
| positions | object | none | Defines the positions of the card elements. See [Positions Options](#positions-options).
| severity | object | none | A list of severity values. See [Severity Options](#severity-options).
| target | number | none | Defines and enables target marker value.
| width | string | 100% | Defines the width of the bar (**Required** the name must be on `outside`).

## Severity Options

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| from | number | **Required** | Defines from which value the color should be displayed.
| to | number | **Required** | Defines to which value the color should be displayed.
| color | string | **Required** | Defines the color to be displayed.

## Positions Options

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| icon | string | outside | `inside`, `outside`, `off`
| name | string | inside | `inside`, `outside`, `off`
| minmax | string | off | `inside`, `outside`, `off`
| value | string | inside | `inside`, `outside`, `off`

```yaml
type: custom:body-miscale-card
entity: bodymiscale.test
model: false
show_name: true
show_states: true
show_attributes: true
show_toolbar: true
show_body: true
show_buttons: true
entity_row: true
buttons:
  user1:
    show: true
body:
  bmi:
    positions:
      name: outside
      minmax: inside
    width: 50%
    target: 21
  bmi_label:
    color: blue
    height: 40px
  visceral_fat:
    severity:
      - from: 0
        to: 4.99
        color: blue
      - from: 5
        to: 10
        color: green
      - from: 10.01
        to: 15
        color: red
    target: 15
    min: 0
    max: 25
```

Translations: Automatic (setting of your homeassistant) or manual
Currently the languages available are `DE`,`EN`,`FR`,`IT`,`NL`,`PT-BR`,`ZH-HANS`, you can contact me to integrate your native language

```yaml
- type: custom:body-miscale-card
  entity: bodymiscale.name
  body:
    water:
      label: 'Eau: '
    bmi:

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## v5.9.4

- added default colors according to the theme

## v5.9.3

- added always show details

## v5.9.2

- added Polish language support (thank you @LukaszP2)

## v5.9.1

- added Romania language support (thank you @18rrs)

## v5.9.0

- add the choice of the theme of your system. (Dark or light)
- added Spanish language (thanks to @luisetex84)

## v5.8.1

- fix error report [[#28](https://github.com/dckiller51/lovelace-body-miscale-card/issues/28)]

## v5.8.0

- update card editor to MWC

## v5.7.1

- fixed de/en syntax (thanks to @tispokes)
- fixed syntax error in Russian (thanks to @dmamontov)

## v5.7.0

- added Russian language (thanks to @BrainFixer)

## v5.6.9

- added Portuguese language (thanks to @jonhdimagio)
- Minor correction in Czech translation

## v5.6.8

- added Czech language (thanks to @xpavli44)

## v5.6.7

- fix: Icons not working in HA 2021.11

## v5.6.6

- Break. Delete webp icons and replace them with png icons to make them compatible with IOS and Safari browser. Attention you must replace your library in www/images/bodyscoreIcon/*.png

## v5.6.5

- added IT language (thanks to @Altar82)

## v5.6.4

- fix CN language to zh-Hans

## v5.6.3

- added CN language (thanks to @sasukebinbin)

## v5.6.2

- fix: Change of target for bone mass

## v5.6.1

- fix: change default values for lbs

## v5.6.0

- Add a switch to convert kg to lbs

## v5.5.0

- Partial integration of bar card options for the body part. (thanks to the contributors of bar-card <https://github.com/custom-cards/bar-card> )
  Options = Partial (Does not work = animation, columns, decimal, complementary, direction)
                    (work = color, height, max, min, positions, severity, target, width)
  Severity Options = OK (from, to, color)
  Positions Options = OK (icon, name, minmax, value (inside, outside, off))

  To know the start, destination, color and target values, open your Mi Fit app on your smartphone.

## v5.0.0

- Conversion to Typescript language (Lit 2)
- Adding the number format according to your system language
- In case of a sensor error the alert icon flashes red
- If no sensor problem the alert icon disappears
- Added chevron-up and chevron-down icons to show or hide measurement details

## v4.6.1

- fix

## v4.6.0

- added NL language (thanks to @llewy)
- update of the elements and translation of the configuration map
- replace the model choice with a switch (Warning: if you upgrade from 4.5.0 to 4.6.0, you must activate a return trip to take into account )

**special thanks to @typxxi for his help in developing the new update and for his extensive testing.**

## v4.5.0

- add BMI Label

## v4.1.1

- fix

## v4.1.0

- added DE language
- replace the icons of materialdesignicons by the official "mifit" icons for the score body

## v4.0.0

- use snake_case format for attribute names (thanks to Pavel Popov <https://github.com/dckiller51/bodymiscale/pull/13>)
- replace custom icons with materialdesignicons icons
- add a toolbar for score and service icons

## v3.1.0

- complete translation of card elements
- added pt-BR language

## v3.0.3

- fix

## v3.0.2

- rename the version with 'v'

## 3.0.1

- fix

## 3.0.0

- added card configuration support
- added automatic translation EN/FR (please contribute to add your language)

## 2.1.0

- Click on the icon to show or hide your score

## 2.0.0

- add the states (on the left)
- move the attributes of the body score "body" (on the bottom)

## 1.0.1

- add the attributes of the body score "body" (for the moment on the left)
example customize:
body:
  water:
    label: 'Water: '

- add the customize icon

## 1.0.0

- **Initial release**


src/const.ts
import { localize } from './localize/localize';

export const CARD_VERSION = '5.9.4';

export const states = {
  status: {
    key: 'status',
    icon: 'mdi:scale-bathroom',
  },
  problem: {
    key: 'problem',
    icon: 'mdi:alert',
  },
};

export const attributes_kg = {
  weight: {
    key: 'weight',
    label: localize(`attributes.${'weight: '}`),
    unit: ' kg',
  },
  impedance: {
    key: 'impedance',
    label: localize(`attributes.${'impedance: '}`),
    unit: ' ohm',
  },
  height: {
    key: 'height',
    label: localize(`attributes.${'height: '}`),
    unit: ' cm',
  },
  age: {
    key: 'age',
    label: localize(`attributes.${'age: '}`),
    unit: localize(`unit.${' years'}`),
  },
  gender: {
    key: 'gender',
    label: localize(`attributes.${'gender: '}`),
  },
};

export const attributes_lb = {
  weight: {
    key: 'weight',
    label: localize(`attributes.${'weight: '}`),
    unit: ' lbs',
  },
  impedance: {
    key: 'impedance',
    label: localize(`attributes.${'impedance: '}`),
    unit: ' ohm',
  },
  height: {
    key: 'height',
    label: localize(`attributes.${'height: '}`),
    unit: ' cm',
  },
  age: {
    key: 'age',
    label: localize(`attributes.${'age: '}`),
    unit: localize(`unit.${' years'}`),
  },
  gender: {
    key: 'gender',
    label: localize(`attributes.${'gender: '}`),
  },
};

export const body_kg = {
  bmi: {
    key: 'bmi',
    label: localize(`body.${'bmi'}`),
    icon: '/local/images/bodyscoreIcon/bmi.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 40,
    min: 0,
    severity: [
      { from: 0, to: 18.5, color: 'blue' },
      { from: 18.51, to: 25, color: 'green' },
      { from: 25.01, to: 28, color: 'orange' },
      { from: 28.01, to: 32, color: 'orangered' },
      { from: 32.01, to: Infinity, color: 'red' },
    ],
    target: 21.75,
  },
  bmi_label: {
    key: 'bmi_label',
    label: localize(`body.${'bmi_label'}`),
    icon: '/local/images/bodyscoreIcon/body_type.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
  },
  metabolic_age: {
    key: 'metabolic_age',
    label: localize(`body.${'metabolic_age'}`),
    icon: '/local/images/bodyscoreIcon/metabolic_age.png',
    unit: localize(`unit.${' years'}`),
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: '',
    min: '',
    severity: '',
    target: '',
  },
  visceral_fat: {
    key: 'visceral_fat',
    label: localize(`body.${'visceral_fat'}`),
    icon: '/local/images/bodyscoreIcon/visceral_fat.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 20,
    min: 0,
    severity: [
      { from: 0, to: 10, color: 'green' },
      { from: 10.01, to: 15, color: 'orange' },
      { from: 15.01, to: Infinity, color: 'orangered' },
    ],
    target: 12.5,
  },
  body_fat: {
    key: 'body_fat',
    label: localize(`body.${'body_fat'}`),
    icon: '/local/images/bodyscoreIcon/body_fat.png',
    unit: ' %',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 40,
    min: 0,
    severity: [
      { from: 0, to: 12, color: 'blue' },
      { from: 12.01, to: 18, color: 'royalblue' },
      { from: 18.01, to: 23, color: 'green' },
      { from: 23.01, to: 28, color: 'orange' },
      { from: 28.01, to: Infinity, color: 'orangered' },
    ],
    target: 20.5,
  },
  protein: {
    key: 'protein',
    label: localize(`body.${'protein'}`),
    icon: '/local/images/bodyscoreIcon/protein.png',
    unit: ' %',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 32,
    min: 0,
    severity: [
      { from: 0, to: 16, color: 'orangered' },
      { from: 16.01, to: 20, color: 'green' },
      { from: 20.01, to: Infinity, color: 'darkgreen' },
    ],
    target: 18,
  },
  water: {
    key: 'water',
    label: localize(`body.${'water'}`),
    icon: '/local/images/bodyscoreIcon/water.png',
    unit: ' %',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 80,
    min: 0,
    severity: [
      { from: 0, to: 55, color: 'orangered' },
      { from: 55.01, to: 65.1, color: 'green' },
      { from: 65.11, to: Infinity, color: 'darkgreen' },
    ],
    target: 60,
  },
  muscle_mass: {
    key: 'muscle_mass',
    label: localize(`body.${'muscle_mass'}`),
    icon: '/local/images/bodyscoreIcon/muscle_mass.png',
    unit: ' kg',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 100,
    min: 0,
    severity: [
      { from: 0, to: 49.4, color: 'orangered' },
      { from: 49.41, to: 59.5, color: 'green' },
      { from: 59.51, to: Infinity, color: 'darkgreen' },
    ],
    target: 54.45,
  },
  bone_mass: {
    key: 'bone_mass',
    label: localize(`body.${'bone_mass'}`),
    icon: '/local/images/bodyscoreIcon/bone_mass.png',
    unit: ' kg',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 8,
    min: 0,
    severity: [
      { from: 0, to: 2, color: 'orangered' },
      { from: 2.01, to: 4.2, color: 'green' },
      { from: 4.21, to: Infinity, color: 'darkgreen' },
    ],
    target: 3.1,
  },
  weight: {
    key: 'weight',
    label: localize(`body.${'weight'}`),
    icon: '/local/images/bodyscoreIcon/ideal.png',
    unit: ' kg',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 130,
    min: 0,
    severity: [
      { from: 0, to: 57.3, color: 'blue' },
      { from: 57.31, to: 77.4, color: 'green' },
      { from: 77.41, to: 86.7, color: 'orange' },
      { from: 86.71, to: 99.1, color: 'orangered' },
      { from: 99.11, to: Infinity, color: 'red' },
    ],
    target: 67.35,
  },
  ideal: {
    key: 'ideal',
    label: localize(`body.${'ideal'}`),
    icon: '/local/images/bodyscoreIcon/ideal.png',
    unit: ' kg',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 130,
    min: 0,
    severity: [
      { from: 0, to: 57.3, color: 'blue' },
      { from: 57.31, to: 77.4, color: 'green' },
      { from: 77.41, to: 86.7, color: 'orange' },
      { from: 86.71, to: 99.1, color: 'orangered' },
      { from: 99.11, to: Infinity, color: 'red' },
    ],
    target: 67.35,
  },
  basal_metabolism: {
    key: 'basal_metabolism',
    label: localize(`body.${'basal_metabolism'}`),
    icon: '/local/images/bodyscoreIcon/basal_metabolism.png',
    unit: ' kcal',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 3000,
    min: 0,
    severity: [
      { from: 0, to: 1530, color: 'orangered' },
      { from: 1530.01, to: Infinity, color: 'green' },
    ],
    target: 1530,
  },
  body_type: {
    key: 'body_type',
    label: localize(`body.${'body_type'}`),
    icon: '/local/images/bodyscoreIcon/body_type.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
  },
};

export const body_lb = {
  bmi: {
    key: 'bmi',
    label: localize(`body.${'bmi'}`),
    icon: '/local/images/bodyscoreIcon/bmi.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 40,
    min: 0,
    severity: [
      { from: 0, to: 18.5, color: 'blue' },
      { from: 18.51, to: 25, color: 'green' },
      { from: 25.01, to: 28, color: 'orange' },
      { from: 28.01, to: 32, color: 'orangered' },
      { from: 32.01, to: Infinity, color: 'red' },
    ],
    target: 21.75,
  },
  bmi_label: {
    key: 'bmi_label',
    label: localize(`body.${'bmi_label'}`),
    icon: '/local/images/bodyscoreIcon/body_type.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
  },
  metabolic_age: {
    key: 'metabolic_age',
    label: localize(`body.${'metabolic_age'}`),
    icon: '/local/images/bodyscoreIcon/metabolic_age.png',
    unit: localize(`unit.${' years'}`),
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: '',
    min: '',
    severity: '',
    target: '',
  },
  visceral_fat: {
    key: 'visceral_fat',
    label: localize(`body.${'visceral_fat'}`),
    icon: '/local/images/bodyscoreIcon/visceral_fat.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 20,
    min: 0,
    severity: [
      { from: 0, to: 10, color: 'green' },
      { from: 10.01, to: 15, color: 'orange' },
      { from: 15.01, to: Infinity, color: 'orangered' },
    ],
    target: 12.5,
  },
  body_fat: {
    key: 'body_fat',
    label: localize(`body.${'body_fat'}`),
    icon: '/local/images/bodyscoreIcon/body_fat.png',
    unit: ' %',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 40,
    min: 0,
    severity: [
      { from: 0, to: 12, color: 'blue' },
      { from: 12.01, to: 18, color: 'royalblue' },
      { from: 18.01, to: 23, color: 'green' },
      { from: 23.01, to: 28, color: 'orange' },
      { from: 28.01, to: Infinity, color: 'orangered' },
    ],
    target: 20.5,
  },
  protein: {
    key: 'protein',
    label: localize(`body.${'protein'}`),
    icon: '/local/images/bodyscoreIcon/protein.png',
    unit: ' %',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 32,
    min: 0,
    severity: [
      { from: 0, to: 16, color: 'orangered' },
      { from: 16.01, to: 20, color: 'green' },
      { from: 20.01, to: Infinity, color: 'darkgreen' },
    ],
    target: 18,
  },
  water: {
    key: 'water',
    label: localize(`body.${'water'}`),
    icon: '/local/images/bodyscoreIcon/water.png',
    unit: ' %',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 80,
    min: 0,
    severity: [
      { from: 0, to: 55, color: 'orangered' },
      { from: 55.01, to: 65.1, color: 'green' },
      { from: 65.11, to: Infinity, color: 'darkgreen' },
    ],
    target: 60,
  },
  muscle_mass: {
    key: 'muscle_mass',
    label: localize(`body.${'muscle_mass'}`),
    icon: '/local/images/bodyscoreIcon/muscle_mass.png',
    unit: ' lbs',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 220.5,
    min: 0,
    severity: [
      { from: 0, to: 108.9, color: 'orangered' },
      { from: 109, to: 131.2, color: 'green' },
      { from: 131.3, to: Infinity, color: 'darkgreen' },
    ],
    target: 120.1,
  },
  bone_mass: {
    key: 'bone_mass',
    label: localize(`body.${'bone_mass'}`),
    icon: '/local/images/bodyscoreIcon/bone_mass.png',
    unit: ' lbs',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 17.6,
    min: 0,
    severity: [
      { from: 0, to: 4.4, color: 'orangered' },
      { from: 4.5, to: 9.3, color: 'green' },
      { from: 9.4, to: Infinity, color: 'darkgreen' },
    ],
    target: 6.9,
  },
  weight: {
    key: 'weight',
    label: localize(`body.${'weight'}`),
    icon: '/local/images/bodyscoreIcon/ideal.png',
    unit: ' lbs',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 286.6,
    min: 0,
    severity: [
      { from: 0, to: 126.3, color: 'blue' },
      { from: 126.4, to: 170.6, color: 'green' },
      { from: 170.7, to: 191.1, color: 'orange' },
      { from: 191.2, to: 218.5, color: 'orangered' },
      { from: 218.6, to: Infinity, color: 'red' },
    ],
    target: 148.5,
  },
  ideal: {
    key: 'ideal',
    label: localize(`body.${'ideal'}`),
    icon: '/local/images/bodyscoreIcon/ideal.png',
    unit: ' lbs',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 286.6,
    min: 0,
    severity: [
      { from: 0, to: 126.3, color: 'blue' },
      { from: 126.4, to: 170.6, color: 'green' },
      { from: 170.7, to: 191.1, color: 'orange' },
      { from: 191.2, to: 218.5, color: 'orangered' },
      { from: 218.6, to: Infinity, color: 'red' },
    ],
    target: 148.5,
  },
  basal_metabolism: {
    key: 'basal_metabolism',
    label: localize(`body.${'basal_metabolism'}`),
    icon: '/local/images/bodyscoreIcon/basal_metabolism.png',
    unit: ' kcal',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      indicator: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
    max: 3000,
    min: 0,
    severity: [
      { from: 0, to: 1530, color: 'orangered' },
      { from: 1530.01, to: Infinity, color: 'green' },
    ],
    target: 1530,
  },
  body_type: {
    key: 'body_type',
    label: localize(`body.${'body_type'}`),
    icon: '/local/images/bodyscoreIcon/body_type.png',
    direction: 'right',
    height: '30px',
    width: '100%',
    positions: {
      icon: 'outside',
      name: 'inside',
      minmax: 'off',
      value: 'inside',
    },
    color: 'var(--score-card-color, var(--ha-card-background))',
  },
};

export const buttons = {
  user1: {
    show: false,
    label: 'User1',
    icon: 'mdi:alpha-u-circle',
  },
  user2: {
    show: false,
    label: 'User2',
    icon: 'mdi:alpha-u-circle',
  },
  user3: {
    show: false,
    label: 'User3',
    icon: 'mdi:alpha-u-circle',
  },
  user4: {
    show: false,
    label: 'User4',
    icon: 'mdi:alpha-u-circle',
  },
  user5: {
    show: false,
    label: 'User5',
    icon: 'mdi:alpha-u-circle',
  },
};

const compute = {
  convertkgtolb: (v: any) => Math.round((Number(v) * 2.20462) * 10) / 10,
}

export const models = {
  false: {
    states: {
      status: {
        key: 'state',
      },
    },
    attributes_kg: {
      weight: { key: 'weight' },
      impedance: false,
      height: { key: 'height' },
      age: { key: 'age' },
      gender: { key: 'gender' },
    },
    attributes_lb: {
      weight: {
        key: 'weight',
        compute: compute.convertkgtolb,
      },
      impedance: false,
      height: { key: 'height' },
      age: { key: 'age' },
      gender: { key: 'gender' },
    },
    body_kg: {
      bmi: { key: 'bmi' },
      bmi_label: { key: 'bmi_label' },
      visceral_fat: { key: 'visceral_fat' },
      metabolic_age: false,
      body_fat: false,
      protein: false,
      water: false,
      muscle_mass: false,
      bone_mass: false,
      weight: { key: 'weight' },
      ideal: { key: 'ideal' },
      basal_metabolism: { key: 'basal_metabolism' },
      body_type: false,
    },
    body_lb: {
      bmi: { key: 'bmi' },
      bmi_label: { key: 'bmi_label' },
      visceral_fat: { key: 'visceral_fat' },
      metabolic_age: false,
      body_fat: false,
      protein: false,
      water: false,
      muscle_mass: false,
      bone_mass: false,
      weight: {
        key: 'weight',
        compute: compute.convertkgtolb,
      },
      ideal: {
        key: 'ideal',
        compute: compute.convertkgtolb,
      },
      basal_metabolism: { key: 'basal_metabolism' },
      body_type: false,
    },
  },
  true: {
    states: {
      status: {
        key: 'state',
      },
    },
    attributes_kg: {
      weight: { key: 'weight' },
      impedance: { key: 'impedance' },
      height: { key: 'height' },
      age: { key: 'age' },
      gender: { key: 'gender' },
    },
    attributes_lb: {
      weight: {
        key: 'weight',
        compute: compute.convertkgtolb,
      },
      impedance: { key: 'impedance' },
      height: { key: 'height' },
      age: { key: 'age' },
      gender: { key: 'gender' },
    },
    body_kg: {
      bmi: { key: 'bmi' },
      bmi_label: { key: 'bmi_label' },
      metabolic_age: { key: 'metabolic_age' },
      visceral_fat: { key: 'visceral_fat' },
      body_fat: { key: 'body_fat' },
      protein: { key: 'protein' },
      water: { key: 'water' },
      muscle_mass: { key: 'muscle_mass' },
      bone_mass: { key: 'bone_mass' },
      weight: { key: 'weight' },
      ideal: { key: 'ideal' },
      basal_metabolism: { key: 'basal_metabolism' },
      body_type: { key: 'body_type' },
    },
    body_lb: {
      bmi: { key: 'bmi' },
      bmi_label: { key: 'bmi_label' },
      metabolic_age: { key: 'metabolic_age' },
      visceral_fat: { key: 'visceral_fat' },
      body_fat: { key: 'body_fat' },
      protein: { key: 'protein' },
      water: { key: 'water' },
      muscle_mass: {
        key: 'muscle_mass',
        compute: compute.convertkgtolb,
      },
      bone_mass: {
        key: 'bone_mass',
        compute: compute.convertkgtolb,
      },
      weight: {
        key: 'weight',
        compute: compute.convertkgtolb,
      },
      ideal: {
        key: 'ideal',
        compute: compute.convertkgtolb,
      },
      basal_metabolism: { key: 'basal_metabolism' },
      body_type: { key: 'body_type' },
    },
  },
};
